### DISCUSSION TITLE
Requirement about Eloquent Acessor json serialization

### DISCUSSION THREAD

**Discussion by EronAlves1996** [laravel/framework#50432] [Q&A]
Given a model with a `price` field, that maps directly to a database `price` field, I am creating an Eloquent mutator/acessor for it, like this:

```php
protected function price(): Attribute
    {
        return Attribute::make(
            get: fn(int $value) => $value / 100,
            set: fn(float $value) => $value * 100
        );
    }
```

But I found that, when I configure this attribute to append it to json serialization like this:

```php
protected $appends = ['price'];
```

It fails with a exception:

```
"message": "App\\Models\\Product::App\\Models\\{closure}(): Argument #1 ($name) must be of type string, null given",

```

Investigating the reason behind this, finally I found that overriding existing fields of Model with acessor/mutator requires enabling the attributes caching like this:

 ```php
  protected function price(): Attribute
    {
        return Attribute::make(
            get: fn(int $value) => $value / 100,
            set: fn(float $value) => $value * 100
        )->shouldCache();
    }
```

This should be on the docs, it isn't? In [this page](https://laravel.com/docs/10.x/eloquent-mutators), that talks especifically about acessors/mutators on eloquent, and [this page](https://laravel.com/docs/10.x/eloquent-serialization) that talks about eloquent serialization, I dont found any information about pitfalls in appending existing fields acessors to json.

**Comment by shaedrich** [comment]
* As far as I know, overriding existing fields was impossible, using the old method—_should_ it still be?
* Assuming, it _should_, this should not be a matter of documentation but independent of caching.
